src/modules/sharepoint/sharepoint.service.test.ts

```diff
@@ -1,76 +1,218 @@
-import { GraphService } from '@ukef/modules/graph/graph.service';
+import { SharepointResourceTypeEnum } from '@ukef/constants/enums/sharepoint-resource-type';
 import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
-import { when } from 'jest-when';
 
-import { SharepointService } from './sharepoint.service';
+import { SharepointService, SharepointUploadFileParams } from './sharepoint.service';
+import { getMockSharepointConfig } from './sharepoint.test-parts/mock-sharepoint-config.helper';
+import { withGetMethodTests } from './sharepoint.test-parts/with-get-method-tests';
+import { withPatchMethodTests } from './sharepoint.test-parts/with-patch-method-tests';
+import { withPostMethodTests } from './sharepoint.test-parts/with-post-method-tests';
+import { withUploadFileMethodTests } from './sharepoint.test-parts/with-upload-file-method-tests';
 
 jest.mock('@ukef/modules/graph/graph.service');
 
 describe('SharepointService', () => {
   const valueGenerator = new RandomValueGenerator();
 
-  let graphServiceGet: jest.Mock;
+  const siteId = valueGenerator.ukefSiteId();
+  const buyerName = valueGenerator.buyerName();
+  const exporterName = valueGenerator.exporterName();
+  const marketName = valueGenerator.string();
+  const dealFolderName = valueGenerator.string();
+  const facilityIdentifier = valueGenerator.facilityId();
+  const sharepointResourceType = valueGenerator.enumValue(SharepointResourceTypeEnum) as SharepointResourceTypeEnum;
```

Comment:
Minor point, but the intended usage here if you need to give the typing a kick is `valueGenerator.enumValue<SharepointResourceTypeEnum>(SharepointResourceTypeEnum);` instead of `as SharepointResourceTypeEnum`

Reply:
This will be solved in a later PR